nixos/modules/hardware/video/nvidia.nix

```diff
@@ -4,8 +4,10 @@
   pkgs,
   ...
 }: let
+  x11Enabled = config.services.xserver.enable
+               && (lib.elem "nvidia" config.services.xserver.videoDrivers);
   nvidia_x11 =
-    if (lib.elem "nvidia" config.services.xserver.videoDrivers)
+    if  x11Enabled || cfg.datacenter.enable
```

Comment:
This would previously make the nvidia driver available without setting `services.xserver.enable` to `true`. I was using this with sway successfully. With this PR sway fails because there is no nvidia driver. I agree that activating the nvidia driver using `services.xserver.videoDrivers` is weird but this change breaks stuff without giving us an option to activate the driver without also activating X11. :slightly_frowning_face:

Reply:
Right, sorry about that! but the fix is pretty simple. Just change `x11Enabled` to not switch on `services.xserver.enable`

Reply:
Addressed in #254978